guardian/recipes-backend
# Telemetry for launch & takedown events

## What does this change?

Sends messages to https://github.com/guardian/data-science-recipes/pull/83 when recipes are launched and taken down

## How to test

- Deploy to CODE
- Make sure that the `/CODE/content-api/recipes-responder/telemetryXAR` and `/TEST/content-api/recipes-responder/telemetryTopic` SSM properties are set
- Launch a recipe in CODE
- Check the logs. You should see some extra log lines:
  - `INFO Refreshing cross-account credentials for telemetry` (maybe, depending on whether a refresh was needed)
  - `INFO Telemetry message for PublishedData send with message ID ....`

There is a pre-requisite PR on the Data Science side that makes sure the messages are handled correctly: https://github.com/guardian/data-science-recipes/pull/92.  That explains where to find the data

## How can we measure success?

Able to track launched data

## Have we considered potential risks?

n/a
